### DISCUSSION TITLE
Response type naming convention

### DISCUSSION THREAD

**Discussion by zidad** [christianhelle/refitter#544] [Q&A]
Hi, 

Thanks a lot for your great library! I'm playing around with it now, it seems to look quite well, except for the response type naming, I get classes named Response1 ... Response22... 

Is it possible to specify a naming convention based on the operation, if there's no type name defined, like with the operation name?
So something like this:   "responseTypeNameTemplate": "{operationName}Response"?

That would be my only blocker to keep the generated code as-is without having to make manual corrections.

Thanks!

**Comment by christianhelle** [comment]
@zidad Thanks for taking the time to bring this up

As of now, Refitter does not support it. Users have had similar complaints, but I haven't gotten to building the feature yet. Refitter piggybacks on NSwag for parsing the OpenAPI document, and features like this manipulate the NSwag OpenAPI document object before generating Refit code. We already do this for the tag/match limiting feature, so there are references in the code on how to manipulate the NSwag OpenAPI document.

You're more than welcome to give the implementation a shot. I'm very open to pull requests, so if you build it and it works, then I'll make sure we get it merged in and released.

I can also build this myself, but it might take a few weeks since this is not a feature I would use myself, and I'm busy with other stuff

**Reply by zidad** [reply]
I'll see if I can take a look at it!
